internal/venice-common/src/main/java/com/linkedin/venice/meta/MaterializedViewParameters.java

```diff
@@ -0,0 +1,115 @@
+package com.linkedin.venice.meta;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.linkedin.venice.exceptions.VeniceException;
+import com.linkedin.venice.utils.ObjectMapperFactory;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+
+public enum MaterializedViewParameters {
+  /**
+   * Parameter key used to specify the re-partition view name.
+   */
+  MATERIALIZED_VIEW_NAME,
+  /**
+   * Parameter key used to specify the partitioner for the re-partition view.
+   */
+  MATERIALIZED_VIEW_PARTITIONER,
+  /**
+   * Parameter key used to specify the partitioner parameters for the partitioner associated with the re-partition view.
+   */
+  MATERIALIZED_VIEW_PARTITIONER_PARAMS,
+  /**
+   * Parameter key used to specify the partition count for the re-partition view.
+   */
+  MATERIALIZED_VIEW_PARTITION_COUNT,
+
+  /**
+   * Parameter key used to specify the top level fields to perform projection on for records in the materialized view.
+   */
+  MATERIALIZED_VIEW_PROJECTION_FIELDS;
+
+  public static class Builder {
+    private String viewName;
+    private String partitioner;
+    private String partitionerParams;
+    private String partitionCount;
+    private List<String> projectionFields = Collections.emptyList();
+
+    public Builder(String viewName) {
+      this.viewName = Objects.requireNonNull(viewName, "View name cannot be null for ViewParameters");
+    }
+
+    public Builder(String viewName, Map<String, String> viewParams) {
```

Comment:
Just a nitpick: An EnumMap can provide a better performance than the Map<String, String> given that all you store is enums. 
"A specialized [Map](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html) implementation for use with enum type keys. All of the keys in an enum map must come from a single enum type that is specified, explicitly or implicitly, when the map is created. Enum maps are represented internally as arrays. This representation is extremely compact and efficient."

Reply:
Let me try that, I'm not sure if it will work because the corresponding interfaces expect a Map<String, String>. Including external/user facing APIs.

Reply:
Not worth it if it needs more than 20 minutes refactor.